### PR TITLE
tests: add Buffer Conversion test for i128

### PR DESCRIPTION
**Pull request by yjhmelody**
Maybe the logic to read 16 bytes should be extracted from i128/u128

### CODE REVIEW

File: assembly/integer/i128.ts [hunk 1]
```diff
@@ -90,28 +90,50 @@ export class i128 {
...
+      return bigEndian
+        ? i128.fromUint8ArrayBE(<Uint8Array>array)
+        : i128.fromUint8ArrayLE(<Uint8Array>array);
+    } else {
+      throw new TypeError("Unsupported generic type");
+    }
   }
 
+  @inline
   static fromBytesLE(array: u8[]): i128 {
```
Comment (yjhmelody) [inline assembly/integer/i128.ts:110]: The original implementation cannot pass the tests

File: assembly/integer/i128.ts [hunk 2]
```diff
@@ -90,28 +90,50 @@ export class i128 {
...
-    var lo: u64 = 0, hi: i64 = 0;
-    for (let i = 0; i <  8; ++i) lo |= unchecked(array[i]) << (i << 3);
-    for (let i = 8; i < 16; ++i) hi |= unchecked(array[i]) << (i << 3);
-
-    return new i128(lo, hi);
+    return i128.fromUint8ArrayLE(changetype<Uint8Array>(array));
   }
 
+  @inline
   static fromBytesBE(array: u8[]): i128 {
```
Comment (yjhmelody) [inline assembly/integer/i128.ts:115]: The original implementation cannot pass the tests

**Comment by yjhmelody** [comment]
OMG, are you already updating this one, I seem to be able to close it

**Comment by MaxGraey** [comment]
thanks!

**Pull request merged**